### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by prateekbh** [developit/preact-mdl#24]
Hey!
Documentation is something really needed here.
reading index.js for props is not really awesome!
do we have something in mind for documentation?
a microsite? another readme?

#HappyToHelp

**Comment by developit** [comment]
I'd love to set at least a readme up, listing the available components and their props.  It might be possible to generate that from the JSDoc annotations already in the code.  Typically that'd be done via [documentation.js](http://documentation.js.org), but I'm open to anything really!

**Comment by prateekbh** [comment]
Yess, that'd be a good start point, but a microsite with code samples is really really helpful.
just a small thing with gh-pages

**Comment by developit** [comment]
yup, can do. the static site generator from preact-www would work, or even just gh-pages' readme default.

**Comment by brenapp** [comment]
Is there an ETA on this? I've had to go back into the React Ecosystem b/c of the lack of documentation on this project. I'd love to be of assistance, but looking through the source files can be pretty unintuitive

**Comment by prateekbh** [comment]
Ahhh its super bad on my part to be honest, @developit shall we quickly check documentation.js and get something rolling out?

**Comment by developit** [comment]
Alright I just quickly generated some (script now runs during build, `npm run docs`). The markdown output isn't great, but it's better than nothing I guess:
https://github.com/developit/preact-mdl/blob/master/docs.md

**Comment by prateekbh** [comment]
Looks good

**Comment by developit** [comment]
lol it exists, wouldn't say it's overly useful. I was hoping for nicer output, but you have to go HTML for any configuration options there.

**Comment by brenapp** [comment]
Examples would be very helpful as well, which I could help with, if you so desire?

**Comment by prateekbh** [comment]
Sure, please raise a PR..
